src/Tuura/Boolean.hs

```diff
@@ -0,0 +1,96 @@
+module Tuura.Boolean (
+  module Tuura.Boolean.Parser,
+  CNF (..), DNF (..), Literal (..),
+  convertToCNF, genConcepts,
+  simplifyCNF, simplifyDNF, convertCNFtoDNF) where
+
+import Data.List
+import Data.List.Extra
+import Data.Foldable
+import Data.Maybe
+import Data.Ord
+
+import Tuura.Boolean.Parser
+import Tuura.Concept.Circuit
+
+newtype CNF a = CNF { fromCNF :: [[Literal a]] }
+
+newtype DNF a = DNF { fromDNF :: [[Literal a]] }
+
+data Literal a = Literal { variable :: a, polarity :: Bool } deriving (Eq, Ord, Show)
+
+convertToCNF :: Eq a => Expr a -> CNF a
+convertToCNF expr = cnf
+  where
+    vars = nub $ toList expr
+    values = mapM (const [False, True]) vars
+    fs = filter (not . eval expr . getValue vars) values
+    sim = CNF $ map (\v -> nub $ map (\f -> Literal (vars !! f) (v !! f)) [0..(length vars - 1)]) fs
```

Comment:
I think this can be simplified:
* `nub` is probably redundant?
* By inserting `not` before `v !! f` you immediately get the `cnf` -- no need to invert `sim` in the next line.